app/controllers/CheckYourAnswersController.scala

```diff
@@ -61,12 +64,32 @@ class CheckYourAnswersController @Inject()(identify: IdentifierAction,
   }
 
   def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-
     submissionService.createCase.map {
-      case Right(value) => Ok(confirmationView(value.id))
+      case Right(value) =>
+        (for {
+          entryDetails <- request.userAnswers.get(EntryDetailsPage)
+          eoriDetails <- request.userAnswers.get(KnownEoriDetails)
+          importerName <- Some(request.userAnswers.get(ImporterNamePage).getOrElse(eoriDetails.name))
+          eoriNumber <- Some(request.userAnswers.get(ImporterEORINumberPage).getOrElse(eoriDetails.eori))
+        } yield {
+          val formattedDate = entryDetails.entryDate.format(DateTimeFormatter.ofPattern("dd/MM/uuuu"))
+          ConfirmationViewData(
+            s"${entryDetails.epu}-${entryDetails.entryNumber}-${formattedDate}",
+            importerName,
+            eoriNumber
+          )
+        }) match {
```

Comment:
Reading this for a second time it might be clearer if you assign the output of the for-yield to a meaningful variable such as confirmationData before doing a match on that variable. Breaking it into 2 separate statement will lower the complexity measure of this method.